pkg/bgpv1/gobgp/server.go

```diff
@@ -91,17 +98,66 @@ func NewGoBGPServerWithConfig(ctx context.Context, log *logrus.Entry, params typ
 // AddNeighbor will add the CiliumBGPNeighbor to the gobgp.BgpServer, creating
 // a BGP peering connection.
 func (g *GoBGPServer) AddNeighbor(ctx context.Context, n types.NeighborRequest) error {
+	peer, err := g.getPeerForNeighbor(ctx, n.Neighbor, false)
+	if err != nil {
+		return err
+	}
+	peerReq := &gobgp.AddPeerRequest{
+		Peer: peer,
+	}
+	if err = g.server.AddPeer(ctx, peerReq); err != nil {
+		return fmt.Errorf("failed while adding peer %v %v: %w", n.Neighbor.PeerAddress, n.Neighbor.PeerASN, err)
+	}
+	return nil
+}
+
+// UpdateNeighbor will update the existing CiliumBGPNeighbor in the gobgp.BgpServer.
+func (g *GoBGPServer) UpdateNeighbor(ctx context.Context, n types.NeighborRequest) error {
+	peer, err := g.getPeerForNeighbor(ctx, n.Neighbor, true)
+	if err != nil {
+		return err
+	}
+	peerReq := &gobgp.UpdatePeerRequest{
+		DoSoftResetIn: true, // should perform soft reset only if needed
+		Peer:          peer,
+	}
+	if _, err = g.server.UpdatePeer(ctx, peerReq); err != nil {
+		return fmt.Errorf("failed while updating peer %v %v: %w", n.Neighbor.PeerAddress, n.Neighbor.PeerASN, err)
+	}
+	return nil
+}
+
+// getPeerForNeighbor returns GoBGP Peer configuration for the provided CiliumBGPNeighbor.
+func (g *GoBGPServer) getPeerForNeighbor(ctx context.Context, n *v2alpha1api.CiliumBGPNeighbor, isUpdate bool) (*gobgp.Peer, error) {
```

Comment:
nit: method name is bothering me little :) getPeerConfig would be better description of what is going on in it.

Reply:
agree - naming is hard :)